codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerServiceGeneratorV2.kt

```diff
@@ -224,9 +233,28 @@ class ServerServiceGeneratorV2(
             "NullabilityChecks" to nullabilityChecks,
             "RoutesArrayElements" to routesArrayElements,
             "SmithyHttpServer" to smithyHttpServer,
+            "PatternInitializations" to patternInitializations(),
         )
     }
 
+    /**
+     * Renders `PatternString::compile_regex()` function calls for every
+     * `@pattern`-constrained string shape in the service closure.
+     */
+    @Suppress("DEPRECATION")
+    private fun patternInitializations(): Writable =
+        Walker(model).walkShapes(service).mapNotNull { shape ->
```

Comment:
Can't you do a `filter` and then a `map` instead of a `mapNotNull` here?

Reply:
I tend to intentionally use `mapNotNull` over `filter + map` for efficiency (iterating the list once). It should be called `filterMap` to make the intent clearer : p.

I don't feel strongly about this, so happy to change it.

Reply:
Done in 4213e8840bcb2ff802d1ec30d8c414ac895178aa.